src/stream_processor/CMakeLists.txt

```diff
@@ -1,7 +1,45 @@
 cmake_minimum_required(VERSION 2.8)
 project(stream-processor)
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+include_directories(
+  ../../lib/onigmo/
```

Comment:
Is the `../../lib/onigmo/` change needed here?

Reply:
onigmo.h does not exist in CMAKE_CURRENT_SOURCE_DIR in first building.

Reply:
why ${CMAKE_CURRENT_SOURCE_DIR} is not working ?

Reply:
Because onigmo will be copied from its source directory:
https://github.com/fluent/fluent-bit/blob/master/cmake/onigmo.cmake#L41-L42

Before building and installing onigmo, stream processor code cannot refer onigmo.h.

Reply:
Genrated cmake file(DependInfo.cmake) in build/src/stream_processor does not contain original `onigmo.h` location:

```cmake
# The include file search paths:
set(CMAKE_C_TARGET_INCLUDE_PATH
  "../include"
  "../lib"
  "../lib/flb_libco"
  "../lib/msgpack-3.1.1/include"
  "../lib/chunkio/include"
  "../lib/LuaJIT-2.1.0-beta3/src"
  "../lib/monkey/include"
  "../lib/mbedtls-2.14.1/include"
  "../lib/sqlite-amalgamation-3240000"
  "backtrace-prefix/include"
  "include"
  "../src/."
  "../src/../lib"
  "../src/../lib/jsmn"
  "../src/../lib/msgpack-c-0b7cabd/include"
  "../src/../lib/monkey/include"
  "../src/../lib/luajit-2.0.5/src"
  "../src/stream_processor"
  )
```

then, `cmake ..` and `make -j5` causes build failure:

```
checking whether gcc accepts -g... [ 15%] Building C object src/stream_processor/parser/CMakeFiles/flb-sp-parser.dir/sql_lex.c.o
[ 15%] Building C object lib/mbedtls-2.14.1/library/CMakeFiles/mbedcrypto.dir/chacha20.c.o
yes
checking for gcc option to accept ISO C89... [ 16%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/cio_stream.c.o
[ 16%] Building C object lib/mbedtls-2.14.1/library/CMakeFiles/mbedcrypto.dir/chachapoly.c.o
none needed
checking for style of include used by make... GNU
checking dependency style of gcc... [ 16%] Building C object lib/chunkio/src/CMakeFiles/chunkio-static.dir/chunkio.c.o
In file included from /Users/cosmo/GitHub/fluent-bit/src/stream_processor/parser/flb_sp_parser.c:30:
In file included from /Users/cosmo/GitHub/fluent-bit/include/fluent-bit/stream_processor/flb_sp_parser.h:26:
In file included from /Users/cosmo/GitHub/fluent-bit/include/fluent-bit/stream_processor/flb_sp.h:27:
In file included from /Users/cosmo/GitHub/fluent-bit/include/fluent-bit/flb_input.h:33:
In file included from /Users/cosmo/GitHub/fluent-bit/include/fluent-bit/flb_filter.h:27:
/Users/cosmo/GitHub/fluent-bit/include/fluent-bit/flb_regex.h:33:10: fatal error: 'onigmo.h' file not found
#include <onigmo.h>
         ^~~~~~~~~~
```

This error is caused by parallel building.

Reply:
I noticed that this issue is occurred in flb_regex.c
This issue caused by parallel building race condition.
I'll remove https://github.com/fluent/fluent-bit/pull/1215/commits/8d77d730ee4801a55ae7e429da17510a6fa423ea commit.
This should be treated as another issue.

Reply:
I've registered [an another PR](https://github.com/fluent/fluent-bit/pull/1226) to fix this issue.